addons/mail/models/mail_scheduled_message.py

```diff
@@ -177,39 +178,46 @@ def _post_message(self, raise_exception=True):
             This is useful when scheduled messages are sent from the _post_messages_cron.
         """
         notification_parameters_whitelist = self._notification_parameters_whitelist()
+        auto_commit = not getattr(threading.current_thread(), 'testing', False)
```

Comment:
This will work greatly, but wouldn't it be better to patch `cr.commit` and `cr.rollback` in the `setUpClass` instead of adding test-related conditions to the code?
That could look like this:
```python
    with patch.object(cls.env.cr, 'commit', side_effect=cls.env.flush_all), \
         patch.object(cls.env.cr, 'rollback', return_value=None)
```
Not sure about what is the policy for that in standard Odoo, maybe @tde-banana-odoo knows better?

Reply:
Since, you need to wait for the rd-security :seal: of approval :wink: